app/controllers/admin/web_hooks_controller.rb

```diff
@@ -92,6 +92,38 @@ def redeliver_event
     end
   end
 
+  def ping
+    uri = URI(@web_hook.payload_url)
+    conn = Excon.new(uri.to_s,
+                     ssl_verify_peer: @web_hook.verify_certificate,
+                     retry_limit: 0)
+
+    body = 'ping'
+    content_type = case @web_hook.content_type
+                   when WebHook.content_types['application/x-www-form-urlencoded']
+                     'application/x-www-form-urlencoded'
+                   else
+                     'application/json'
+                   end
+    headers = {
+      'Accept' => '*/*',
+      'Connection' => 'close',
+      'Content-Length' => body.size,
+      'Content-Type' => content_type,
+      'Host' => uri.host,
+      'User-Agent' => "Discourse/" + Discourse::VERSION::STRING,
+      'X-Discourse-Event-Id' => 'null',
+      'X-Discourse-Event-Type' => 'ping'
+    }
+
+    if @web_hook.secret.present?
+      headers['X-Discourse-Event-Signature'] = "sha256=" + OpenSSL::HMAC.hexdigest("sha256", @web_hook.secret, body)
+    end
+    conn.post(headers: headers, body: body)
```

Comment:
If it isn't functional.. How will they know whether it is their configuration or a bug in the code?